### PR TITLE
New Solid Refresh

### PR DESCRIPTION
**Pull request by lxsmnsyc**
- Unifies API for `standard`, `esm`, `vite` and `webpack5`
- Adds support for Context API
- Simplifies generated code
- Drop support for anonymous `export default` components
- Now uses `exports` map (~~breaks [vite-plugin-solid](https://github.com/solidjs/vite-plugin-solid/blob/master/src/index.ts)~~)
- Fix dependency generation to skip if empty
- Switched from SHA-256 to [js-xxhash](https://github.com/Jason3S/xxhash) for signature generation
- New `location` and solid-refresh naming scheme for solid-devtools (cc @thetarnav)
- Fix granular mode dependencies to be out-of-order
- Fix proper top-level decline for non-ESM HMR

solves #33, #24, #15, #9

### CODE REVIEW

**Comment by lxsmnsyc** [comment]
Also needs E2E testing specially on Vite and Rollup

**Comment by lxsmnsyc** [comment]
Known issues (checked = solved):
- [ ] Parcel loads production build of SolidJS, thus failing to resolve properly for Solid Refresh
- [ ] Parcel fails to load Solid Refresh through the JSON form of Babel
- [ ] `@babel/generator` behaves differently between `esm` and `cjs` formats

**Comment by lxsmnsyc** [comment]
Due to some issues with Parcel, I decided to just drop its support. Users still using Parcel should use the current version of SolidJS

File: src/runtime/index.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  id: string,
+  component: (props: P) => JSX.Element,
+  options: ComponentOptions = {},
+): (props: P) => JSX.Element {
+  if (!component.name) {
+    setComponentProperty(component, 'name', id);
+  }
+  if (options.location) {
+    setComponentProperty(component, 'location', options.location);
+  }
```
Comment (thetarnav) [inline src/runtime/index.ts:59]: Actually, I'm not sure I have a way to access this function? 🤔
The `owner.fn` of proxy component would be the proxy
And `owner.fn` of the hmr memo is [this](https://github.com/solidjs/solid-refresh/blob/c98cb9ad830dab8621d7cac1e61f3ef9d3f3ab59/src/runtime/create-proxy.ts#L32)
So wouldn't it be better to add the location/name to the `get` traps of the proxy component?

Comment (lxsmnsyc) [reply src/runtime/index.ts:59]: @thetarnav I agree that `createComponent` should be modified for that purpose.

**Pull request merged**